atd-etl/cr3_download/cr3_download.py

```diff
@@ -5,145 +5,186 @@
+        "-v", "--verbose", action="store_true", help="Enable verbose logging"
     )
+    args = parser.parse_args()
 
-    print("\nStarting CR3 downloads:")
-except Exception as e:
-    crashes_list_without_skips = []
-    print("Error, could not run CR3 processing: " + str(e))
+    # Start timer
+    start = time.time()
+
+    # Get 1Password secrets from environment
+    ONEPASSWORD_CONNECT_HOST = os.getenv("OP_CONNECT")
+    ONEPASSWORD_CONNECT_TOKEN = os.getenv("OP_API_TOKEN")
+    VAULT_ID = os.getenv("OP_VAULT_ID")
+
+    # Setup 1Password server connection
+    one_password_client = new_client(
+        ONEPASSWORD_CONNECT_HOST, ONEPASSWORD_CONNECT_TOKEN
+    )
 
+    # Load secrets from 1Password and set them in the environment
+    env_vars = load_secrets(one_password_client, VAULT_ID)
+    for key, value in env_vars.items():
+        os.environ[key] = value
 
-for crash_record in crashes_list_without_skips:
-    process_crash_cr3(
-        crash_record,
-        CRIS_BROWSER_COOKIES,
-        skipped_uploads_and_updates,
+    # Ask user for a set of valid cookies for requests to the CRIS website
+    CRIS_BROWSER_COOKIES = input(
+        "Please login to CRIS and extract the contents of the Cookie: header and please paste it here: "
     )
 
-print("\nProcess done.")
+    if args.verbose:
+        print("Preparing download loop.")
+        print("Gathering list of crashes.")
+
+    # Track crash IDs that we don't successfully retrieve a pdf file for
+    skipped_uploads_and_updates = []
+
+    # Some crash IDs were manually added at the request of the VZ team so
+    # CR3s for these crash IDs are not available in the CRIS database.
+    # We can skip requesting them.
+    # See https://github.com/cityofaustin/atd-data-tech/issues/9786
+    known_skips = [180290542, 144720068]
+
+    crashes_list_without_skips = []
+
+    try:
+        if args.verbose:
+            print(f"Hasura endpoint: '{os.getenv('HASURA_ENDPOINT')}'")
+
+        response = get_crash_id_list()
+
+        crashes_list = response["data"]["atd_txdot_crashes"]
+        crashes_list_without_skips = [
+            x for x in crashes_list if x["crash_id"] not in known_skips
+        ]
+
+    except Exception as e:
+        crashes_list_without_skips = []
+        print(f"Error, could not run CR3 processing: {str(e)}")
+
+    if args.verbose:
+        print(f"Length of queue: {len(crashes_list_without_skips)}")
+        print("Starting CR3 downloads:")
+
+    max_workers = args.threads
+    with ThreadPoolExecutor(max_workers=max_workers) as executor:
+        futures = []
+        for crash_record in crashes_list_without_skips:
+            future = executor.submit(
+                process_crash_cr3_threaded,
+                crash_record,
+                CRIS_BROWSER_COOKIES,
+                skipped_uploads_and_updates,
+                args.verbose,
+            )
+            futures.append(future)
+
+        for future in futures:
+            future.result()
```

Comment:
This is the meat of the changes. Almost all of this is linting / or changing the indention / block level.